crates/polars-plan/src/dsl/function_expr/strings.rs

```diff
@@ -466,6 +466,7 @@ fn extract_many(
     ascii_case_insensitive: bool,
     overlapping: bool,
 ) -> PolarsResult<Column> {
+    _check_same_length(s, "extract_many")?;
```

Comment:
Was this change really necessary?
This change broke the R API by preventing the use of str type Series of longer than length 2 as input.